### ISSUE TITLE
.NET Core 3.0 preview8 ?

### ISSUE THREAD

**Issue by dotnetshadow** [MiniProfiler/dotnet#412]
Hi there,

I've tried to upgrade my asp.net core 2.2 application to asp.net core 3.0 preview8 and having an issue similar to #376

I did try with asp.net core 3.0 preview7 and had the same issue
Just wondering if this should be working?

Using version 4.0.180

```cs
An unhandled exception occurred while processing the request.
TypeLoadException: Could not load type 'Microsoft.AspNetCore.Mvc.ViewFeatures.Internal.ViewComponentInvokerCache' from assembly 'Microsoft.AspNetCore.Mvc.ViewFeatures, Version=3.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'.

System.Signature.GetSignature(Void* pCorSig, int cCorSig, RuntimeFieldHandleInternal fieldHandle, IRuntimeMethodInfo methodHandle, RuntimeType declaringType)

    TypeLoadException: Could not load type 'Microsoft.AspNetCore.Mvc.ViewFeatures.Internal.ViewComponentInvokerCache' from assembly 'Microsoft.AspNetCore.Mvc.ViewFeatures, Version=3.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'.
```

**UPDATE:**
I also tried one of your later builds from MyGet **MiniProfiler.AspNetCore.Mvc 4.1.0-preview.12**
but encountered the following issue

```cs
info: Microsoft.AspNetCore.Hosting.Diagnostics[1]
      Request starting HTTP/2 GET https://localhost:5101/mini-profiler-resources/includes.min.css?v=4.1.0-preview.12+a7322be1d9
info: Microsoft.AspNetCore.Hosting.Diagnostics[2]
      Request finished in 1.1369ms 200 text/css
fail: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware[1]
      An unhandled exception has occurred while executing the request.
System.TypeLoadException: Could not load type 'System.Text.Json.Serialization.JsonSerializer' from assembly 'System.Text.Json, Version=4.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'.
   at StackExchange.Profiling.MiniProfilerMiddleware.GetSingleProfilerResultAsync(HttpContext context)
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at StackExchange.Profiling.MiniProfilerMiddleware.GetSingleProfilerResultAsync(HttpContext context)
   at StackExchange.Profiling.MiniProfilerMiddleware.HandleRequest(HttpContext context, PathString subPath) in C:\projects\dotnet\src\MiniProfiler.AspNetCore\MiniProfilerMiddleware.cs:line 217
   at StackExchange.Profiling.MiniProfilerMiddleware.Invoke(HttpContext context) in C:\projects\dotnet\src\MiniProfiler.AspNetCore\MiniProfilerMiddleware.cs:line 74
```

**Comment by TheArchitectIO** [comment]
I am having the same exact issues

I believe it has to do with [this file](https://github.com/MiniProfiler/dotnet/blob/master/src/MiniProfiler.AspNetCore.Mvc/MvcExtensions.cs#L37) 

```
#if !NETCOREAPP3_0
            services.AddTransient<IViewComponentInvokerFactory, ProfilingViewComponentInvokerFactory>();
#endif
```

Reproducible CS

```
namespace FailingMiniProfiler
{
    using Microsoft.AspNetCore.Builder;
    using Microsoft.AspNetCore.Hosting;
    using Microsoft.Extensions.Configuration;
    using Microsoft.Extensions.Hosting;
    using Microsoft.Extensions.DependencyInjection;

    /*
     * <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="3.0.0-preview8.19405.7" />
     * <PackageReference Include="MiniProfiler.AspNetCore.Mvc" Version="4.0.180" />
     */

    public class Program
    {
        public static void Main(string[] args)
        {
            CreateHostBuilder(args).Build().Run();
        }

        public static IHostBuilder CreateHostBuilder(string[] args) =>
            Host.CreateDefaultBuilder(args)
                .ConfigureWebHostDefaults(webBuilder =>
                    {
                        webBuilder.UseStartup<Startup>();
                    });
    }

    public class Startup
    {
        public Startup(IConfiguration configuration)
        {
            Configuration = configuration;
        }

        public IConfiguration Configuration { get; }

        // This method gets called by the runtime. Use this method to add services to the container.
        public void ConfigureServices(IServiceCollection services)
        {
            services
                .AddControllers()
                .AddNewtonsoftJson();

            services.AddMiniProfiler();
        }

        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
        {
            if (env.IsDevelopment())
            {
                app.UseDeveloperExceptionPage();
            }

            app.UseHttpsRedirection();
            app.UseMiniProfiler();
            app.UseRouting();

            app.UseAuthorization();

            app.UseEndpoints(endpoints =>
            {
                endpoints.MapControllers();
            });
        }
    }
}

```

**Comment by NickCraver** [comment]
There was a breaking issue (namespace changes in System.Text.Json) between preview 7 and preview 8.

I've just pushed a new `4.1.0-preview.21` build [to MyGet](https://www.myget.org/gallery/miniprofiler) - see if that resolves your issue?

**Comment by TheArchitectIO** [comment]
I tried this morning and all looks to be working with preview8 of and 4.1.0-preview.21. I am no longer personally getting `Could not load type 'System.Text.Json.Serialization.JsonSerializer'`

Thanks you guys are awesome as always

**Comment by dotnetshadow** [comment]
Can confirm that this is working as well, thanks for the great work

**Comment by marchy** [comment]
Sorry for the dumb question, but how can you reference the MyGet package from a .csproj?

Latest version that shows up on Nuget is 4.0.180 (including if 'Prerelease' packages are enabled)

**Comment by NickCraver** [comment]
You'll need a `nuget.config` file in your repo root, which [looks like this](https://docs.microsoft.com/en-us/nuget/reference/nuget-config-file) and you'll want the feed URL here: https://www.myget.org/gallery/miniprofiler (click "connect to feed").

As a complete example, here's Opserver's main config file which allow usage of all Stack Overflow pre-release packages (I publish MiniProfiler to that feed and its own): https://github.com/opserver/Opserver/blob/master/nuget.config
```xml
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <clear /> <!-- Optional - ensures no system sources for a clean build the same everywhere -->
    <add key="MyGet: Stack Overflow" value="https://www.myget.org/F/stackoverflow/api/v3/index.json" />
    <add key="NuGet" value="https://api.nuget.org/v3/index.json" />
  </packageSources>
</configuration>
```

This repo-based approach should work in all IDEs.

**Comment by NickCraver** [comment]
I'm going to close this our since it's available, but once 3.0 ships (should be Monday) I'll follw-up quickly release a MiniProfiler build depending on non-preview packages for 3.0.
